network/dag/bbolt_dag.go

```diff
@@ -154,6 +154,23 @@ func (dag bboltDAG) GetByPayloadHash(payloadHash hash.SHA256Hash) ([]Transaction
 	return result, err
 }
 
+func (dag *bboltDAG) PayloadHashes(visitor func(payloadHash hash.SHA256Hash) error) error {
+	return dag.db.View(func(tx *bbolt.Tx) error {
+		payloadIndex := tx.Bucket([]byte(payloadIndexBucket))
```

Comment:
How is this bucket structured. Is it payloadHash => payload or txRef => payload or?

Reply:
so it's a reverse index ;)